flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/TaskManagerLocation.java

```diff
@@ -150,28 +166,48 @@ public String addressString() {
 
 	/**
 	 * Returns the fully-qualified domain name the TaskManager. If the name could not be
-	 * determined, the return value will be a textual representation of the TaskManager's IP address.
+	 * determined or {@link JobManagerOptions.RETRIEVE_TASK_MANAGER_HOSTNAME} is set to false,
```

Comment:
Isn't the use of this configuration option an implementation of the `JobMaster`, not visible to this class?
I would only mention the `retrieveHostName` field here.

Reply:
You are right that the implementation details of JobMaster should not be exposed to other classes, and I have updated the JavaDocs.